pmlc/rt/runtime_registry.h

```diff
@@ -75,8 +72,27 @@ struct LoaderRegistration {
 template <class R>
 struct RuntimeRegistration {
```

Comment:
Please call this `FactoryRegistration`, since it's registering a factory. I'd prefer that we had a `RuntimeRegistration` that just registered a runtime directly.

Reply:
To clarify, I'd expect to see:

```
template <class R>
struct FactoryRegistration {
  explicit RuntimeRegistration(llvm::StringRef id) {
    registerFactory(id, makeStaticFactory<R>());
  }
};

struct RuntimeRegistration {
  explicit RuntimeRegistration(llvm::StringRef id, std::shared_ptr<Runtime> runtime) {
    registerRuntime(id, runtime);
  }
};
```

Reply:
I'd rather hide the whole factory thing from components that use this if possible.  Also, we can't actually call registerRuntime() at static initialization time, because we want the exception to be caught by `initRuntimes()`.  So, how about this?

Reply:
OK, LGTM